decidim-core/app/jobs/decidim/delete_inactive_participants_job.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module Decidim
+  # A job to delete inactive participants and notify them
+  class DeleteInactiveParticipantsJob < ApplicationJob
+    queue_as :delete_inactive_participants
+
+    def perform(organization)
+      clear_notifications_for_active_users(organization)
+      send_first_warning_notifications(organization.users.first_warning_inactive_users)
+      send_last_warning_notifications(organization.users.last_warning_inactive_users)
+      remove_inactive_users(organization.users.removable_users)
+    end
+
+    private
+
+    def clear_notifications_for_active_users(organization)
+      process_users(organization.users.not_deleted
+                                .with_inactivity_notification
+                                .active_after_notification) do |user|
+        user.extended_data.delete("inactivity_notification")
+        user.save!
```

Comment:
I do have a problem banging this save. If there is a number of user to which you have to remove the notification, and one has a validation issue, all the rest will be ignored. I would extract this in a job.